include/fiction/algorithms/simulation/sidb/critical_temperature.hpp

```diff
@@ -410,18 +410,28 @@ class critical_temperature_impl
     [[nodiscard]] sidb_simulation_result<Lyt>
     physical_simulation_of_layout(const bdl_input_iterator<Lyt>& bdl_iterator) noexcept
     {
-        assert(params.physical_parameters.base == 2 && "base number has to be 2");
-
-        if (params.engine == critical_temperature_params::simulation_engine::EXACT)
+        if (params.engine == sidb_simulation_engine::QUICKEXACT)
         {
-            // perform exact simulation
+            assert(params.physical_parameters.base == 2 && "base number has to be 2");
+
+            // perform QuickExact simulation
             const quickexact_params<Lyt> qe_params{
                 params.physical_parameters, fiction::quickexact_params<Lyt>::automatic_base_number_detection::OFF};
             return quickexact(*bdl_iterator, qe_params);
         }
 
-        if (params.engine == critical_temperature_params::simulation_engine::APPROXIMATE)
+#if (FICTION_ALGLIB_ENABLED)
+        if (params.engine == sidb_simulation_engine::CLUSTERCOMPLETE)
+        {
+            // perform ClusterComplete simulation -- base 3 simulation is allowed
```

Comment:
I would strongly recommend running ``can_positive_charges_occur`` first. If it returns ``true``, the CT should be zero. The reason for this is: 1.) In general, we have less information about the physics when it comes to positively charged SiDBs. 2.) If we add the temperature, I am even more concerned, because this is already a strong approximation and we should not overdo it here. Maybe something we can also add as a ``note`` in the docstring?

Reply:
My reasoning is just from the computational model, so thanks for complementing it with your understanding of the physics! 
I think a note would suffice, but without the CT=0K remedy I would suggest, as that limits experimentation. Perhaps even a warning should be printed when positive charges can occur, but you would leave the rest to the model as is. The user may then understand that this area of CT is pre-alpha so to say. 

In any case, I think a note should be good to add that the temperature model is an approximation, since that is not clear.

Maybe you could formulate a little note like that plus perhaps a warning message for when positive charges can occur? I'm not aware of how the temperature model is limited here.

Big thanks for the new review round by the way!!

Reply:
Sure, sounds really good! I will add something. Thank you!